modules/packages/LinearAlgebra.chpl

```diff
@@ -1331,6 +1331,100 @@ proc solve (A: [?Adom] ?eltType, b: [?bdom] eltType) {
   return x;
 }
 
+/* Compute least-squares solution to ``A * x = b``.
+   Compute a vector ``x`` such that the 2-norm ``|b - A x|`` is minimized.
+
+   ``cond`` is the cut-off threshold such that singular values will be
+   considered 0.0. If ``cond < 0.0`` (defaults to ``-1.0``), the treshold will
+   be set to ``max((...A.shape)) * epsilon``, where ``epsilon`` is the machine
+   precision for ``A.eltType``.
+
+   Returns a tuple of ``(x, residues, rank, s)``, where:
+
+  - ``x`` is the the least-squares solution with shape of ``b``
+  - ``residues`` is the square of the 2-norm for each column in ``b - a x`` if ``M > N`` and ``A.rank == n``
+    - ``residues`` is a scalar if ``b`` is 1-D, otherwise it is a 1-element array
+  - ``rank`` is the effective rank of ``A``
+  - ``s`` is the singular values of ``a``
+
+    .. note::
+
+      This procedure depends on the :mod:`LAPACK` module, and will generate a
+      compiler error if ``lapackImpl`` is ``none``.
+*/
+proc leastSquares(A: [] ?t, b: [] t, cond = -1.0) throws
+  where A.rank == 2 && b.rank == 1 && usingLAPACK && isLAPACKType(t)
+{
+  use SysCTypes;
+  import LAPACK;
+  require LAPACK.header;
+
+
+  if A.shape[0] != b.shape[0] {
+    throw new LinearAlgebraError('leastSquares(): A.shape[0] != b.shape[0]: %i != %i'.format(A.shape[0], b.shape[0]));
+  }
+  if A.size == 0 || b.size == 0 {
+    throw new LinearAlgebraError('leastSquares(): A and b cannot be empty');
+  }
+
+  // TODO: Support b.rank == 2  -- (m,k)
+  const (m, n) = A.shape;
+
+  // TODO: Support overwrite=true/false
+  var workA = A;
+  const bdim1 = b.domain.dim(0);
+  const bdim2 = if b.rank == 1 then 0..0
+                else b.domain.dim(1);
+
+  var workB: [0..<b.size, 0..<bdim2.size] real;
+  if b.rank == 1 then
+    workB[.., 0] = b;
+  else
+    workB = b;
+
+  const rcond = if cond == -1.0 then max((...A.shape))*epsilon(t) else cond;
+
+  var matrix_order = lapack_memory_order.row_major;
+
+  var s: [0..<min((...A.shape))] real(64);
+  var rank: c_int;
+
+  var info = LAPACK.gelsd(matrix_order, workA, workB, s, rcond, rank);
+
+  // Check for errors
+  if info < 0 then
+    throw new owned IllegalArgumentError('gelsd(): Argument %i incorrect'.format(info));
+  else if info > 0 then
+    throw new owned LinearAlgebraError('gelsd(): SVD failed to converge with %i off-diagonal elements not converged to 0'.format(info));
+
+  var x1 = workB[0..<n, 0];
```

Comment:
I can't find what `n` is, but I am probably missing something

Reply:
1370: `const (m, n) = A.shape;`